internal/controlplane/luascripts/remove-impersonate-headers.lua

```diff
@@ -0,0 +1,25 @@
+local function starts_with(str, start)
+    return str:sub(1, #start) == start
+end
+
+function envoy_on_request(request_handle)
+    local headers = request_handle:headers()
+    local metadata = request_handle:metadata()
+
+    local remove_impersonate_headers = metadata:get("remove_impersonate_headers")
+    if remove_impersonate_headers then
+        local to_remove = {}
```

Comment:
according to the docs we aren't allowed to modify the header collection while looping over it